### PR TITLE
🎉 Source Cockroachdb: Added new source for Cockroachdb

### LINKED ISSUE #1705: Support CockroachDB as a new source connector
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368084037114) by [Unito](https://www.unito.io)

Use postgresql jdbc driver and refer postgres source.

### PR DESCRIPTION
**Pull request by etsybaev**
## What
Added CockroachDB source connector.

## How
Added CockroachDB source connector.
Some of data type returns wrong values, but those issues are same for all of DB. A follow-up ticket's been created https://github.com/airbytehq/airbyte/issues/4408

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [x] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [x] Documentation updated 
    - [x] `README.md`
    - [x] `docs/SUMMARY.md` if it's a new connector
    - [x] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md` contains a reference to the new connector
    - [x] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [x] Build is successful
- [x] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [x] No major blockers
- [x] PR merged into master branch
- [x] Follow up tickets have been created
- [x] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

Selection_191
Selection_190
Selection_189
Selection_188
Selection_187
Selection_186

### CODE REVIEW

File: airbyte-integrations/connectors/source-cockroachdb/src/main/java/io/airbyte/integrations/source/cockroachdb/CockroachdbSource.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import io.airbyte.protocol.models.AirbyteConnectionStatus;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CockroachdbSource extends AbstractJdbcSource implements Source {
```
[the remark this answers is not in the document]
Comment (etsybaev) [reply airbyte-integrations/connectors/source-cockroachdb/src/main/java/io/airbyte/integrations/source/cockroachdb/CockroachdbSource.java]: done

File: airbyte-integrations/connectors/source-cockroachdb/src/test-integration/java/io/airbyte/integrations/source/cockroachdb/CockroachSourceAcceptanceTest.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+import io.airbyte.protocol.models.Field;
+import io.airbyte.protocol.models.JsonSchemaPrimitive;
+import io.airbyte.protocol.models.SyncMode;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import org.jooq.SQLDialect;
+import org.testcontainers.containers.CockroachContainer;
+
+public class CockroachSourceAcceptanceTest extends SourceAcceptanceTest {
```
[the remark this answers is not in the document]
Comment (etsybaev) [reply airbyte-integrations/connectors/source-cockroachdb/src/test-integration/java/io/airbyte/integrations/source/cockroachdb/CockroachSourceAcceptanceTest.java]: done

File: airbyte-integrations/connectors/source-cockroachdb/src/test/java/io/airbyte/integrations/source/cockroachdb/CockroachdbSourceTest.java [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+      final Set<AirbyteMessage> actualMessages = MoreIterators
+          .toSet(new CockroachdbSource().read(config, CONFIGURED_CATALOG, null));
+      setEmittedAtToNull(actualMessages);
+
+      assertEquals(UTF8_MESSAGES, actualMessages);
+    }
+  }
+
+  private static void setEmittedAtToNull(Iterable<AirbyteMessage> messages) {
+    for (AirbyteMessage actualMessage : messages) {
```
[the remark this answers is not in the document]
Comment (etsybaev) [reply airbyte-integrations/connectors/source-cockroachdb/src/test/java/io/airbyte/integrations/source/cockroachdb/CockroachdbSourceTest.java]: done

File: airbyte-integrations/connectors/source-jdbc/build.gradle [hunk 4]
```diff
@@ -51,6 +54,9 @@ dependencies {
     testFixturesImplementation 'org.junit.jupiter:junit-jupiter-params:5.4.2'
     testFixturesImplementation group: 'org.mockito', name: 'mockito-junit-jupiter', version: '3.4.6'
 
+
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-jdbc/build.gradle:57]: remove whitespace

File: airbyte-integrations/connectors/source-jdbc/src/testFixtures/java/io/airbyte/integrations/source/jdbc/test/JdbcSourceAcceptanceTest.java [hunk 5]
```diff
@@ -786,7 +787,7 @@ private ConfiguredAirbyteCatalog getConfiguredCatalogWithOneStream(final String
     return catalog;
   }
 
-  private AirbyteCatalog getCatalog(final String defaultNamespace) {
+  protected AirbyteCatalog getCatalog(final String defaultNamespace) {
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-jdbc/src/testFixtures/java/io/airbyte/integrations/source/jdbc/test/JdbcSourceAcceptanceTest.java:790]: why are these changed to protected? we should keep private unless required to change

File: docs/integrations/sources/cockroachdb.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+| `numeric` | number |  |
+| `smallint` | integer |  |
+| `text` | string |  |
+| `time with timezone` | string | may be written as a native date type depending on the destination |
+| `time without timezone` | string | may be written as a native date type depending on the destination |
+| `timestamp with timezone` | string | may be written as a native date type depending on the destination |
+| `timestamp without timezone` | string | may be written as a native date type depending on the destination |
+| `uuid` | string |  |
+
+**Note:** arrays for all the above types as well as custom types are supported, although they may be de-nested depending on the destination. Byte arrays are currently unsupported.
```
Comment (sherifnada) [inline docs/integrations/sources/cockroachdb.md:38]: what happens to byte arrays? does the connector just break?

File: docs/integrations/sources/cockroachdb.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+### Features
+
+| Feature | Supported | Notes |
+| :--- | :--- | :--- |
+| Full Refresh Sync | Yes |  |
+| Incremental - Append Sync | Yes |  |
```
Comment (sherifnada) [inline docs/integrations/sources/cockroachdb.md:45]:
```suggestion
| Incremental Sync| Yes |  |
| Change Data Capture | No |  |
```

File: airbyte-integrations/connectors/source-cockroachdb/build.gradle [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    integrationTestJavaImplementation project(':airbyte-integrations:connectors:source-cockroachdb')
+    integrationTestJavaImplementation project(':airbyte-integrations:bases:standard-source-test')
+
+    implementation files(project(':airbyte-integrations:bases:base-java').airbyteDocker.outputs)
+    integrationTestJavaImplementation files(project(':airbyte-integrations:bases:base-java').airbyteDocker.outputs)
+
+    testImplementation testFixtures(project(':airbyte-integrations:connectors:source-jdbc'))
+    testImplementation project(":airbyte-json-validation")
+    testImplementation project(':airbyte-test-utils')
+
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-cockroachdb/build.gradle:34]: remove whitespace

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/connectors/source-jdbc/src/testFixtures/java/io/airbyte/integrations/source/jdbc/test/JdbcSourceAcceptanceTest.java (hunk 5 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-jdbc/src/testFixtures/java/io/airbyte/integrations/source/jdbc/test/JdbcSourceAcceptanceTest.java:790]: We need to override it in CockroachDbJdbcSourceAcceptanceTest. Basic Catalog contains a case with the table without PK meanwhile the Cockroach DB is not supposed to have a tables without PK. If you do not set it explicitly - it will anyway automatically add additionally.

File: docs/integrations/sources/cockroachdb.md (hunk 6 above)
Comment (etsybaev) [reply docs/integrations/sources/cockroachdb.md:38]: My bad, I copy-pasted the template from the postgres source and probably missed to clean this Note. Byte arrays also works, added additional test for it to comprehensive tests. Thanks

File: airbyte-integrations/connectors/source-jdbc/build.gradle (hunk 4 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-jdbc/build.gradle]: done, thanks

File: docs/integrations/sources/cockroachdb.md (hunk 7 above)
Comment (etsybaev) [reply docs/integrations/sources/cockroachdb.md]: done, thanks

File: airbyte-integrations/connectors/source-cockroachdb/build.gradle (hunk 8 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/source-cockroachdb/build.gradle]: done, thanks

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml [hunk 9]
```diff
@@ -49,6 +49,12 @@
   dockerImageTag: 0.3.4
   documentationUrl: https://hub.docker.com/r/airbyte/source-postgres
   icon: postgresql.svg
+- sourceDefinitionId: 9fa5862c-da7c-11eb-8d19-0242ac130003
+  name: Cockroachdb
+  dockerRepository: airbyte/source-postgres
+  dockerImageTag: 0.1.0
+  documentationUrl: https://hub.docker.com/r/airbyte/source-cockroachdb
+  icon: cockroachdb.svg
```
Comment (sherifnada) [inline airbyte-config/init/src/main/resources/seed/source_definitions.yaml:57]: there is currently a bug that causes airbyte to break when new icons are added. Let's remove the field for `icon` (but keep the icon itself in the repo) until this bug is fixed.

**Review by sherifnada — APPROVED** [review]
LGTM -- can you make sure to update all the docs mentioned in the checklist? some of them are not updated

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml (hunk 9 above)
Comment (etsybaev) [reply airbyte-config/init/src/main/resources/seed/source_definitions.yaml]: Updated, thanks for letting me know
